lms/services/blackboard_api/client.py

```diff
@@ -169,7 +169,9 @@ def course_groups(
             responses = self._request.find_service(name="async_oauth_http").request(
                 "GET", self_enrollment_check_urls
             )
-            for group, response in zip(self_enrollment_groups, responses):
+            # We are going to be modifying `self_enrollment_groups` within the loop
+            # Create a copy here to iterate over.
+            for group, response in zip(list(self_enrollment_groups), responses):
```

Comment:
I haven't been able to write good test case that didn't pass for the previous version. I'll play a little with it later and do a follow up PR.